### PR TITLE
feature: Domain added notification

### PR DESCRIPTION
**Pull request by izzy64**
#  #

## 🗣 Description ##

When new sending profiles are added to Con-PCA, active customers will get an email prompting them to add the new domain to their safelist.

This goal prompted an additional change, a small refactoring of the notifications utility. The Notification() function was made more flexible and generic: a cycle is now an optional argument, and additional custom arguments can now be passed in when required.

## 💭 Motivation and context ##

This PR represents a feature request from the client. 

In the past, keeping customer safelists consistent with our collection of sending domains has required significant effort from the admins. This change attempts to automate part of that process, by sending an email to all active primary contacts whenever a new sending profile is added to the system. This email contains the name of the new domain to be added to the customers' safelist.

## 🧪 Testing ##

These changes were tested manually, by adding a new sending profile to a local version of the application. 

More testing will be added soon, as we implement unit testing throughout the project. 

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Add a tag or create a release.

### CODE REVIEW

File: src/api/templates/emails/domain_added_notice.html [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+                          <div
+                            style="
+                              font-family: helvetica;
+                              font-size: 20px;
+                              line-height: 1;
+                              text-align: left;
+                              color: #1b2352;
+                            "
+                          >
+                            A new domain has been added to our systems. To
```
Comment (itsmostafa) [inline src/api/templates/emails/domain_added_notice.html]: Please change the first sentence to "A new sending domain has been added."

**Review by itsmostafa — CHANGES_REQUESTED** [review]

File: src/utils/notifications.py [hunk 2]
```diff
@@ -97,6 +113,10 @@ def get_report(self, message_type: str, context: dict):
                 "to": "primary_contact",
                 "bcc": "admin",
             },
+            "domain_added_notice": {
+                "subject": "Con-PCA Update to Safelisting Domains",
```
Comment (itsmostafa) [inline src/utils/notifications.py]: Please change the Subject line to "Con-PCA New Sending Domain Added"

**Review by itsmostafa — CHANGES_REQUESTED** [review]

File: src/api/templates/emails/domain_added_notice.html (hunk 1 above)
Comment (izzy64) [reply src/api/templates/emails/domain_added_notice.html]: @itsmostafa changed as requested with ea67d89694fc5d1b924e713049be3bcef582aced

File: src/utils/notifications.py (hunk 2 above)
Comment (izzy64) [reply src/utils/notifications.py]: @itsmostafa changed as requested with ea67d89694fc5d1b924e713049be3bcef582aced

**Review by itsmostafa — APPROVED** [review]

**Pull request merged**